### ISSUE TITLE
Issue with GPU Computation Execution in CUDA Mode

### ISSUE THREAD

**Issue by xiaothua** [mfem/mfem#4374]
Hello,

I have been working on a CUDA program based on the example provided in ex1p and ex14p. However, it seems that while the GPU is being initialized, the computations are still being executed on the CPU.

Compared to the MPI version, I only added `Device device(device_config);` in the main code and defined the device of CUDA. I also recompiled the code in `MFEM_USE_CUDA = YES`.

Could you please advise if there are additional modifications required in the code after selecting the device?

Thank you for your assistance.

Best regards,
Xiaotian

**Comment by v-dobrev** [comment]
Take a look at the "Device sample runs" section in the beginning of the source file -- they represent ways to run the code utilizing primarily the GPU.

Also, if using a parallel MFEM build, I recommend building HYPRE with GPU support as well, see https://mfem.org/building/#cuda-version-of-mfem.

**Comment by xiaothua** [comment]
> Take a look at the "Device sample runs" section in the beginning of the source file -- they represent ways to run the code utilizing primarily the GPU.
> 
> Also, if using a parallel MFEM build, I recommend building HYPRE with GPU support as well, see https://mfem.org/building/#cuda-version-of-mfem.

Thank you for your reply! The information from the link was very helpful. I confirmed that I followed the correct steps for installation and compilation, and other CUDA programs are also running normally.

I tried the "Device sample runs" section and modified my code accordingly. I compared the computation times for choosing GPU versus CPU. However, the computation time for choosing GPU was slightly longer than the latter, so I suspect that the computation part of my code is not running on the GPU.

If it's only necessary to add `const char *device_config = "cpu";` and `Device device(device_config);` at the code level, then perhaps the code is not the issue. I should continue to look for problems hidden in other aspects.
